include/seqan3/io/sam_file/detail/format_sam_base.hpp

```diff
@@ -430,346 +436,291 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
         read_field(string_buffer, value);
     };
 
-    // Some tags are not parsed individually. Instead, these are simply copied into a string.
+    // Some tags are not parsed individually. Instead, these are simply copied into a std::string.
     // Multiple tags must be separated by a `\t`, hence we prepend a tab to the string, except the first time.
     // Alternatively, we could always append a `\t`, but this would have the side effect that we might need to trim a
     // trailing tab after parsing all tags via `pop_back()`.
     // Unfortunately, we do not know when we are parsing the last tag (and in this case just not append a tab),
     // because even though we can check if the line ends in a `\n`, it is not guaranteed that the last tag of the
     // line is passed to this lambda. For example, the line might end with a tag that is properly parsed, such as `ID`.
-    auto parse_until_tab_or_newline = [&] (std::string & value, char const first_letter)
+    auto parse_until_tab_or_newline = [&] (std::string & value, std::array<char, 2> raw_tag)
```

Comment:
this name is somehow wrong, because it adds the raw_tag, it seems like a function that is used only in error cases?

Reply:
I think I dont understand what you mean.
This function is used to parse tags plus their value into a string, as these tags are not handled properly. So we parse these tags until their end (tab odr newline) into this string of kind of unhandled tags.

Reply:
I think the name is too general as it does more than that. Why should this lambda append additional stuff from the raw_tag if it is named parse_until_tab_or_newline.

I think something like `parse_unknown_tag_value` would be more appropriate, but I'm not too sure when this function is called, so I need your help what this function actually is intended for.